src/sentry/integrations/middleware/hybrid_cloud/parser.py

```diff
@@ -75,19 +73,20 @@ def __init__(self, request: HttpRequest, response_handler: ResponseHandler):
     # Common Helpers
 
     def ensure_control_silo(self):
-        if SiloMode.get_current_mode() != SiloMode.CONTROL:
-            metrics.incr(
-                self._METRIC_FAILURE_KEY + ".silo_error",
-                sample_rate=1.0,
-                tags={"path": self.request.path, "silo": SiloMode.get_current_mode().value},
-            )
-            logger.error(
-                "ensure_control_silo_error",
-                extra={"path": self.request.path, "silo": SiloMode.get_current_mode()},
-            )
-            raise SiloLimit.AvailabilityError(
-                "Integration Request Parsers should only be run on the control silo."
+        with MiddlewareOperationEvent(
+            operation_type=MiddlewareOperationType.ENSURE_CONTROL_SILO,
+            integration_name=self.provider,
+        ).capture() as lifecycle:
+            lifecycle.add_extras(
+                {
+                    "path": self.request.path,
+                    "silo": str(SiloMode.get_current_mode().value),
```

Comment:
does `SiloMode.get_current_mode()` not resolve to a string? SiloMode has a `__str__` function